### PR TITLE
Automate domain in pv samples

### PR DESCRIPTION
**Pull request by sankarpn**
Automating the domain in pv samples using the wlst and wdt method of creating domains in pv.
Test Results

https://build.weblogick8s.org:8443/job/weblogic-kubernetes-operator-kind-new/2578/testReport/

### CODE REVIEW

File: integration-tests/src/test/java/oracle/weblogic/kubernetes/ItSamples.java [hunk 1]
```diff
@@ -0,0 +1,305 @@
...
+import static oracle.weblogic.kubernetes.utils.ThreadSafeLogger.getLogger;
+import static org.apache.commons.io.FileUtils.copyDirectory;
+import static org.apache.commons.io.FileUtils.deleteDirectory;
+import static org.awaitility.Awaitility.with;
+import static org.junit.jupiter.api.Assertions.assertDoesNotThrow;
+import static org.junit.jupiter.api.Assertions.assertNotNull;
+import static org.junit.jupiter.api.Assertions.assertTrue;
+
+/**
+ * Tests related to introspectVersion attribute.
```
Comment (vanajamukkara) [inline integration-tests/src/test/java/oracle/weblogic/kubernetes/ItSamples.java:52]: fix description

File: integration-tests/src/test/java/oracle/weblogic/kubernetes/ItSamples.java [hunk 2]
```diff
@@ -0,0 +1,305 @@
...
+   * @param namespaces injected by JUnit
+   */
+  @BeforeAll
+  public static void initAll(@Namespaces(2) List<String> namespaces) {
+    logger = getLogger();
+
+    logger.info("Assign a unique namespace for operator");
+    assertNotNull(namespaces.get(0), "Namespace is null");
+    opNamespace = namespaces.get(0);
+    logger.info("Assign a unique namespace for Introspect Version WebLogic domain");
```
Comment (vanajamukkara) [inline integration-tests/src/test/java/oracle/weblogic/kubernetes/ItSamples.java:86]: fix log message

File: integration-tests/src/test/java/oracle/weblogic/kubernetes/ItSamples.java [hunk 3]
```diff
@@ -0,0 +1,305 @@
...
+    boolean result = Command.withParams(params).execute();
+    assertTrue(result, "Failed to create domain.yaml");
+
+    // run kubectl to create the domain
+    params = new CommandParams().defaults();
+    params.command("kubectl apply -f "
+        + Paths.get(sampleBase.toString(), "weblogic-domains/" + domainName + "/domain.yaml").toString());
+
+    result = Command.withParams(params).execute();
+    assertTrue(result, "Failed to create domain");
```
Comment (vanajamukkara) [inline integration-tests/src/test/java/oracle/weblogic/kubernetes/ItSamples.java:147]:
```suggestion
    assertTrue(result, "Failed to create domain custom resource");
```

File: integration-tests/src/test/java/oracle/weblogic/kubernetes/ItSamples.java [hunk 4]
```diff
@@ -0,0 +1,305 @@
...
+    for (int i = 1; i <= replicaCount; i++) {
+      checkPodReadyAndServiceExists(managedServerPodNamePrefix + i, domainName, domainNamespace);
+    }
+
+    //delete the domain resource
+    params = new CommandParams().defaults();
+    params.command("kubectl delete -f "
+        + Paths.get(sampleBase.toString(), "weblogic-domains/" + domainName + "/domain.yaml").toString());
+    result = Command.withParams(params).execute();
+    assertTrue(result, "Failed to delete domain");
```
Comment (vanajamukkara) [inline integration-tests/src/test/java/oracle/weblogic/kubernetes/ItSamples.java:181]:
```suggestion
    assertTrue(result, "Failed to delete domain custom resource");
```

**Review by vanajamukkara — CHANGES_REQUESTED** [review]
LG, update with Jenkins results.

File: integration-tests/src/test/java/oracle/weblogic/kubernetes/ItSamples.java [hunk 5]
```diff
@@ -0,0 +1,305 @@
...
+    installAndVerifyOperator(opNamespace, domainNamespace);
+  }
+
+  /**
+   * Test domain in pv samples using domains created by wlst and wdt.
+   *
+   * @param model domain name and script type to create domain
+   */
+  @ParameterizedTest
+  @MethodSource("paramProvider")
```
Comment (anpanigr) [inline integration-tests/src/test/java/oracle/weblogic/kubernetes/ItSamples.java:100]: What are the possible values for paramProvider ? Put on the description

File: integration-tests/src/test/java/oracle/weblogic/kubernetes/ItSamples.java [hunk 6]
```diff
@@ -0,0 +1,305 @@
...
+  @DisplayName("Test samples using domain in pv")
```
Comment (anpanigr) [inline integration-tests/src/test/java/oracle/weblogic/kubernetes/ItSamples.java:101]: Can you modify the name of the Class to ItDomainOnPvSample.java

**Review by anpanigr — COMMENTED** [review]
Update the PR description with jenkin results.

File: integration-tests/src/test/java/oracle/weblogic/kubernetes/ItSamples.java (hunk 6 above)
Comment (vanajamukkara) [reply integration-tests/src/test/java/oracle/weblogic/kubernetes/ItSamples.java:101]: We can have all the samples in this It class, no need for individual It class for each sample.

File: integration-tests/src/test/java/oracle/weblogic/kubernetes/ItSamples.java (hunk 1 above)
Comment (sankarpn) [reply integration-tests/src/test/java/oracle/weblogic/kubernetes/ItSamples.java:52]: fixed

File: integration-tests/src/test/java/oracle/weblogic/kubernetes/ItSamples.java (hunk 2 above)
Comment (sankarpn) [reply integration-tests/src/test/java/oracle/weblogic/kubernetes/ItSamples.java:86]: fixed

File: integration-tests/src/test/java/oracle/weblogic/kubernetes/ItSamples.java (hunk 5 above)
Comment (sankarpn) [reply integration-tests/src/test/java/oracle/weblogic/kubernetes/ItSamples.java:100]: added the explanation

File: integration-tests/src/test/java/oracle/weblogic/kubernetes/ItSamples.java (hunk 6 above)
Comment (sankarpn) [reply integration-tests/src/test/java/oracle/weblogic/kubernetes/ItSamples.java:101]: Right, I thought of naming it as ItDomainOnPvSample but like as Vanaja said we can have all samples in the same class.

File: integration-tests/src/test/java/oracle/weblogic/kubernetes/ItSamples.java (hunk 3 above)
Comment (sankarpn) [reply integration-tests/src/test/java/oracle/weblogic/kubernetes/ItSamples.java:147]: fixed

File: integration-tests/src/test/java/oracle/weblogic/kubernetes/ItSamples.java (hunk 4 above)
Comment (sankarpn) [reply integration-tests/src/test/java/oracle/weblogic/kubernetes/ItSamples.java:181]: fixed

File: integration-tests/src/test/java/oracle/weblogic/kubernetes/ItSamples.java (hunk 6 above)
Comment (anpanigr) [reply integration-tests/src/test/java/oracle/weblogic/kubernetes/ItSamples.java:101]: OK, Huiling can add  DomainHomeInImage sample to this class once this PR is merged.

**Review by anpanigr — APPROVED** [review]
LGTM

**Review by vanajamukkara — APPROVED** [review]

**Pull request merged**